src/Integration.Vsix/Analysis/SingleFileMonitor.cs

```diff
@@ -119,6 +119,26 @@ private static void EnsureDirectoryExists(string filePath, IFileSystem fileSyste
             }
         }
 
+        private bool IsDuplicateChangeEvent(string fullPath)
+        {
+            // We're trying to ignore duplicate events by checking the last-write time.
+            // However, the precision of DateTime means that it is possible for separate events
+            // that happen very close together to report the same last-write time. If that happens,
+            // we'll be ignoring a "real" notification.
+            var currentTime = File.GetLastWriteTimeUtc(fullPath);
+            var timeDiff = currentTime - lastWriteTime;
+
+            // We are only ignoring events happening within 10 milliseconds of each other.
+            if (timeDiff.TotalMilliseconds >= 0 && timeDiff.TotalMilliseconds <= 10)
```

Comment:
The 10 millisecond is a bit of a guess, I could also make it a hundred, whichever makes more sense